master/buildbot/test/unit/test_www_hooks_github.py

```diff
@@ -434,8 +566,22 @@ def _prepare_request(event, payload, _secret=None, headers=None):
 
 class TestChangeHookConfiguredWithGitChange(unittest.TestCase):
 
+    @defer.inlineCallbacks
     def setUp(self):
-        self.changeHook = _prepare_github_change_hook(strict=False, github_property_whitelist=["github.*"])
+        self.changeHook = _prepare_github_change_hook(
+            strict=False, github_property_whitelist=["github.*"])
+        self.master = self.changeHook.master
+        self._http = yield fakehttpclientservice.HTTPClientService.getFakeService(
+            self.master,
+            self,
+            'https://api.github.com/repos/defunkt/github',
```

Comment:
We need to use https://api.github.com as base URL. So that we can share the same service.  In your implementation one object per project will be created, which is useless